### PR TITLE
Feature/feature meals crud

### PR DESCRIPTION
**Pull request by xavier-campos-diaz**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        materialDatePicker.addOnPositiveButtonClickListener(this::initializeOnPositiveButtonClickListener);
+    }
+
+    /**
+     * Method responsible for initializing the onPositiveButtonClickListener.
+     * @param selection The selected value
+     */
+    private void initializeOnPositiveButtonClickListener(Object selection) {
+        mealDate.setText(materialDatePicker.getHeaderText());
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java:263]: Codacy Issue found: [When instantiating a SimpleDateFormat object, specify a Locale](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  17
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  -11
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMeal.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPetMeal.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllPetMeals.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMealsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteMeal.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPetMeal.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103&bid=17553604)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        List<Meal> mealList = null;
+        try {
+            mealList = ServiceLocator.getInstance().getMealManagerClient().getAllMealData(accessToken,
+                owner, petName);
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+        }
+        List<Meals> result = new ArrayList<>();
+        for (Meal m:mealList) {
+            result.add(new Meals(m));
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java:152]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        binding.inputMealCal.setText(String.valueOf(meal.getKcal()));
+        updatesDate = false;
+        DateTime mealDate = meal.getMealDate();
+        showMealDate(mealDate);
+        showMealTime(mealDate);
+        initializeEditMealButton();
+        initializeRemoveMealButton();
+    }
+
+    private void showMealDate(DateTime mealDate) {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java:90]: You should comment this methods as well

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMeal.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+public class TrUpdateMeal {
+    private MealManagerService mealManagerService;
+    private User user;
+    private Pet pet;
+    private Meals meal;
+    private String newDate;
+    private boolean updatesDate;
+
+    public TrUpdateMeal(MealManagerService mealManagerService) {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMeal.java:16]: You should comment this class as well

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java [hunk 5]
```diff
@@ -362,6 +362,41 @@ public void deleteEvent(Event event) {
...
+
+        for (Event event : events) {
+            if (event instanceof Meals) {
+                mealEvents.add(event);
+            }
+        }
+        return mealEvents;
+    }
+
+    public void deleteAllMeals() {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java]: You should comment this one

**Review by AlbertPG — APPROVED** [review]
Comment this ones

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  17
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  -11
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMeal.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPetMeal.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllPetMeals.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMealsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteMeal.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPetMeal.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103&bid=17553604)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+        materialDatePicker.addOnPositiveButtonClickListener(this::initializeOnPositiveButtonClickListener);
+    }
+
+    /**
+     * Method responsible for initializing the onPositiveButtonClickListener.
+     * @param selection The selected value
+     */
+    private void initializeOnPositiveButtonClickListener(Object selection) {
+        mealDate.setText(materialDatePicker.getHeaderText());
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java:279]: Codacy Issue found: [When instantiating a SimpleDateFormat object, specify a Locale](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java:151]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainAllPetMeals.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPetMeal.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMealsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteUser.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteMeal.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdateMeal.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPetMeal.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103&bid=17553604)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java (hunk 6 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/EditMealFragment.java:279]: Codacy Issue found: [When instantiating a SimpleDateFormat object, specify a Locale](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java (hunk 7 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/MealManagerAdapter.java:151]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5357103)

**Pull request merged**